## Special Pythagorean Triplet

A Pythagorean Triplet is a set of three natural numbers, $a < b < c$, for which, $a^2 + b^2 = c^2$. For example $3^2 + 4^2 = 5^2$.

There exists exactly one Pythagorean triplet for which $a + b + c = 1000$. Find the product $abc$.

### Methodology

Given two equations using a, b, and c we can do a little math first to narrow our approach on finding the triplet. Using the two equations:

1) $a + b + c = 1000$
2) $a^2 + b^2 = c^2$

We can first see $c = 1000 - (a + b)$. Substituting this value into the second equation yields:

$a^2 + b^2 = (1000 - (a+b))^2 \implies 0 = 1000000 - 2000(a+b) + 2ab$

$\implies 500000 + a(b - 1000) - 1000b = 0 \implies a = \frac{500000 - 1000b}{1000-b}$

Since a and b are positive integers such that $a < b$. Thus we're looking for a value of b less than 500 such that $(500000-1000b)$ is divisible by $(1000 - b)$.

In [1]:
# list to store candidates for b
nums = []

for b in range(500):
    if (500000 - 1000*b) % (1000 - b) == 0:
        nums.append(b)

print(nums)

[0, 200, 375]


Since b is larger than a we're looking for the largest number in the list. Thus we get b = 375, which implies a = 200 and c = 425.